lighthouse-core/runner.js

```diff
@@ -150,38 +137,67 @@ class Runner {
    *
    * @template {LH.Config.Config | LH.Config.FRConfig} TConfig
    * @param {(runnerData: {config: TConfig, driverMock?: Driver}) => Promise<LH.Artifacts>} gatherFn
-   * @param {{config: TConfig, driverMock?: Driver}} options
+   * @param {{config: TConfig, driverMock?: Driver, computedCache: Map<string, ArbitraryEqualityMap>}} options
    * @return {Promise<LH.Artifacts>}
    */
-  static async gatherAndManageArtifacts(gatherFn, options) {
+  static async gather(gatherFn, options) {
     const settings = options.config.settings;
 
-    // Gather phase
     // Either load saved artifacts from disk or from the browser.
-    let artifacts;
-    if (settings.auditMode && !settings.gatherMode) {
-      // No browser required, just load the artifacts from disk.
-      const path = this._getDataSavePath(settings);
-      artifacts = assetSaver.loadArtifacts(path);
-      const requestedUrl = artifacts.URL.requestedUrl;
-
-      if (!requestedUrl) {
-        throw new Error('Cannot run audit mode on empty URL');
-      }
-    } else {
-      artifacts = await gatherFn({
-        config: options.config,
-        driverMock: options.driverMock,
+    try {
+      const runnerStatus = {msg: 'Gather phase', id: 'lh:runner:gather'};
+      log.time(runnerStatus, 'verbose');
+
+      const sentryContext = Sentry.getContext();
+      Sentry.captureBreadcrumb({
+        message: 'Run started',
```

Comment:
FWIW the FR runners should call `Runner.gather` every time, but there is no guarantee that the FR runners themselves will be run. The only question is what happens in `user-flow.js`. Should we consider each flow step an independent "run" from the sentry perspective, or should the entire flow be a single "run".

Reply:
I thought from #13364 that they wouldn't be calling into gather() anymore, since it's kind of redundant (they already define their own gather function)? With multiple gathers, at least, -G and -A will have to be tweaked to work.

Reply:
> I thought from #13364 that they wouldn't be calling into gather() anymore

Using `Runner.gather()` in the FR runners is still useful because it gives us the timing entry, error handling, and the sentry call. As you said, we could add those things to each FR runner.

I did update #13364 to reflect that but didn't announce the change sorry.

> With multiple gathers, at least, -G and -A will have to be tweaked to work.

This is true, but we don't *need* to get rid of the logic entirely if the other stuff in `gather()` is still useful. The issue should only be a problem for user flows which will have it's own save/load artifacts system.